plugins/inputs/turbostat/parser/parser.go

```diff
@@ -0,0 +1,232 @@
+		return nil, nil, err
+	}
+	tags := make(tagMap, 0)
+	fields := make(fieldMap, 0)
+	for i := range values {
+		if columns[i].isIgnored {
+			continue
+		}
+		if columns[i].isTag {
+			if !isTagValue(values[i]) {
+				return nil, nil, fmt.Errorf("invalid tag: %s", values[i])
+			}
+			tags[columns[i].name] = values[i]
+		} else {
+			v, err := strconv.ParseFloat(values[i], 64)
+			if err != nil {
+				return nil, nil, err
+			}
+			if columns[i].multiplier != 0 {
+				v *= columns[i].multiplier
+			}
+			fields[columns[i].name] = v
+		}
+	}
+	if len(fields) == 0 {
+		return nil, nil, errors.New("no value for any field")
+	}
+	return tags, fields, nil
+}
+
+// Creates a column struct from a Turbostat column name.
+func createColumn(s string) column {
+	c := column{}
+	// Split the Turbostat column name into tokens.
+	tokens := make([]string, 0)
+	for _, element := range splitSymbols(s) {
+		tokens = append(tokens, splitKnownTokens(element)...)
+	}
+	for i, token := range tokens {
+		token = strings.ToLower(token)
+		// If a tokens is "mhz", set the multiplier to 1M.
+		if token == "mhz" {
+			c.multiplier = 1e6
+		}
+		// If a token is an abbrebiation, replace it.
+		if replacement, ok := abbreviations[token]; ok {
+			token = replacement
+		}
+		tokens[i] = token
+	}
+	// Build the camel case column name.
+	c.name = strings.Join(tokens, "_")
+	// Ignore the timestamp column.
+	if c.name == "time_of_day_seconds" {
+		c.name = "timestamp"
+		c.isIgnored = true
+	}
+	// If the name matches a tag, mark the column as such.
+	if _, ok := tagNames[c.name]; ok {
+		c.isTag = true
+	}
+	return c
+}
+
+// Splits a string into tokens using non-alphanumeric characters as delimiters.
+// The delimiters are discarded unless they belong to a set of symbols to keep.
+func splitSymbols(s string) []string {
+	tokens := make([]string, 0)
+	i := 0
+	start := i
+	for i, c := range s {
+		if !unicode.IsDigit(c) && !unicode.IsLetter(c) {
+			if start < i {
+				tokens = append(tokens, s[start:i])
+			}
+			if _, ok := keepSymbols[c]; ok {
+				tokens = append(tokens, string(c))
+			}
+			start = i + 1
+		}
+	}
+	if start < len(s) {
+		tokens = append(tokens, s[start:])
+	}
+	return tokens
+}
+
+// Splits an alphanumeric string into tokens, using a list
+// of known tokens to determine boundaries.
+func splitKnownTokens(s string) []string {
+	tokens := make([]string, 0)
+	i := 0
+	start := i
+	for i < len(s) {
+		match := false
+		for _, hint := range knownTokens {
+			if strings.HasPrefix(s[i:], hint) {
+				match = true
+				if start < i {
+					tokens = append(tokens, s[start:i])
+				}
+				tokens = append(tokens, hint)
+				i += len(hint)
+				start = i
+				break
+			}
+		}
+		if !match {
+			i++
+		}
+	}
+	if start < i {
+		tokens = append(tokens, s[start:i])
+	}
+	return tokens
+}
+
+// Returns whether a string represents a tag value or not.
+// Turbostat only use integers and "-".
+func isTagValue(s string) bool {
```

Comment:
I think the name of this function might be a bit misleading and suggest that the passed argument can ONLY be a tag value (or not). Maybe `isProperTagValue` would be more appropriate?

Reply:
I went for `IsValidTagValue`, please let me know if that still does not look good to you :pray: